Source/DafnyLanguageServer.Test/Various/ShutdownTest.cs

```diff
@@ -71,12 +71,21 @@ public class ShutdownTest {
 
       // Wait for the language server to kill itself by waiting until it closes the output stream.
       await process.StandardOutput.ReadToEndAsync();
-      await Task.Delay(400); // Give the process some time to die
 
-      Assert.ThrowsException<ArgumentException>(() => {
-        var languageServer = Process.GetProcessById(int.Parse(languageServerProcessId));
-        languageServer.Kill();
-      }, "Language server should have killed itself if the parent is gone.");
+      for (int attempts = 0; attempts < 10; attempts++) {
+        // Give the language server time to kill itself
+        await Task.Delay(100);
+
+        try {
+          Process.GetProcessById(int.Parse(languageServerProcessId));
+        } catch {
+          return;
+        }
+      }
+
+      var languageServer = Process.GetProcessById(int.Parse(languageServerProcessId));
```

Comment:
Couldn't we do this assignment in the loop?

Reply:
This is the cleanup code once we've decided the language server won't kill itself so we're going to manually do it.

Reply:
Sure, but why the assignment.

Reply:
The value is used on the next line. I'm not sure I understand the question.

Reply:
Yes, but you also generate it in L80. Might as well use that.

Reply:
Okay. I just looked into doing that but it creates warnings around assignment that I'd rather not override.

Reply:
Thanks for looking into it.